aeneas/src/v3/TypeSystem.v3

```diff
@@ -123,6 +135,9 @@ component TypeSystem {
 			return null;
 		} else if (V3.isTuple(t1) && V3.isTuple(t2)) {
 			return mergeLists(t1.nested, t2.nested, Tuple.TYPECON, upper);
+		} else if (V3.isRef(t1) && V3.isRef(t2)) {
```

Comment:
You probably also want to implement a case for `isNullable()`, which will hit above.